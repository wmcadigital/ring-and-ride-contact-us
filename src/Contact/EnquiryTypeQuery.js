import { useFormState } from "react-final-form";

import EnquiryTypes from "./EnquiryTypes";
import FormSection from "../common/FormSection";
import RadioGroup from "../common/RadioGroup";
import Question from "../common/Question";
import RadioButton from "../common/RadioButton";
import FieldError from "../common/FieldError";
import { required } from "../common/validation";

const EnquiryTypeQuery = () => {
  const stateApi = useFormState();

  const error = stateApi.submitFailed ? stateApi.errors?.enquiryType : null;

  return (
    <FormSection>
      <Question text="What is your enquiry about?" />
      <RadioGroup error={error}>
        <FieldError text={error} />
        {Object.keys(EnquiryTypes).map((enquiryType, index) => (
          <RadioButton
            key={index}
            label={EnquiryTypes[enquiryType].radioLabel}
            validation={required}
            value={enquiryType}
            fieldName="enquiryType"
          />
        ))}
      </RadioGroup>
    </FormSection>
  );
};

export default EnquiryTypeQuery;
