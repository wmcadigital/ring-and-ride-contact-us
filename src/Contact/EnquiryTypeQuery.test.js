import { useFormState } from "react-final-form";
import { createRenderer } from "react-test-renderer/shallow";

import EnquiryTypeQuery from "./EnquiryTypeQuery";

jest.mock("react-final-form");

describe("EnquiryTypeQuery", () => {
  it("renders as expected", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        enquiryType: "lostProperty",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<EnquiryTypeQuery />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders with an error", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {},
      submitFailed: true,
      errors: { enquiryType: "Required" },
    }));
    const renderer = createRenderer();

    renderer.render(<EnquiryTypeQuery />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
