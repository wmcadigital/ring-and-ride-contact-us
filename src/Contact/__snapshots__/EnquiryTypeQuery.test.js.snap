// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnquiryTypeQuery renders as expected 1`] = `
<FormSection>
  <Question
    text="What is your enquiry about?"
  />
  <RadioGroup
    error={null}
  >
    <FieldError
      text={null}
    />
    <RadioButton
      fieldName="enquiryType"
      label="General enquiry"
      validation={[Function]}
      value="generalEnquiry"
    />
    <RadioButton
      fieldName="enquiryType"
      label="Service issues"
      validation={[Function]}
      value="serviceIssues"
    />
    <RadioButton
      fieldName="enquiryType"
      label="Lost property"
      validation={[Function]}
      value="lostProperty"
    />
    <RadioButton
      fieldName="enquiryType"
      label="Staff"
      validation={[Function]}
      value="staff"
    />
    <RadioButton
      fieldName="enquiryType"
      label="Service updates"
      validation={[Function]}
      value="serviceUpdates"
    />
  </RadioGroup>
</FormSection>
`;

exports[`EnquiryTypeQuery renders with an error 1`] = `
<FormSection>
  <Question
    text="What is your enquiry about?"
  />
  <RadioGroup
    error="Required"
  >
    <FieldError
      text="Required"
    />
    <RadioButton
      fieldName="enquiryType"
      label="General enquiry"
      validation={[Function]}
      value="generalEnquiry"
    />
    <RadioButton
      fieldName="enquiryType"
      label="Service issues"
      validation={[Function]}
      value="serviceIssues"
    />
    <RadioButton
      fieldName="enquiryType"
      label="Lost property"
      validation={[Function]}
      value="lostProperty"
    />
    <RadioButton
      fieldName="enquiryType"
      label="Staff"
      validation={[Function]}
      value="staff"
    />
    <RadioButton
      fieldName="enquiryType"
      label="Service updates"
      validation={[Function]}
      value="serviceUpdates"
    />
  </RadioGroup>
</FormSection>
`;
